### PR TITLE
Implement Tree Paint

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Based on the work by @marijnvdwerf a few years back.

### CODE REVIEW

File: src/OpenLoco/Windows/TerraForm.cpp [hunk 1]
```diff
@@ -870,10 +869,10 @@ namespace OpenLoco::Ui::Windows::Terraform
 
         static void drawTreeThumb(TreeObject* treeObj, Gfx::Context* clipped)
         {
-            uint32_t image = _byte_500775[treeObj->growth] * treeObj->num_rotations;
+            uint32_t image = treeGrowth[treeObj->growth] * treeObj->num_rotations;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/TerraForm.cpp:872]: I think it's a bit confusing that this is basically a global that's defined in the tree object header. Perhaps it's better to expose the tree growth array through a member function?

File: src/OpenLoco/Paint/PaintTree.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+#include "../Ui.h"
+#include "Paint.h"
+
+using namespace OpenLoco::Interop;
+using namespace OpenLoco::Ui::ViewportInteraction;
+
+namespace OpenLoco::Paint
+{
+    constexpr std::array<uint8_t, 6> _50076A = { 3, 0, 1, 2, 1, 4 };
+    constexpr std::array<bool, 6> _500770 = { true, true, false, false, true, true };
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Paint/PaintTree.cpp:16]: These are used by a `season` variable. Any idea why there are six elements?

Also, the values from `_50076A` are used to index into `_500770`. Any idea why are there six elements? The last one is unreachable, isn't it?

Comment (duncanspumpkin) [reply src/OpenLoco/Paint/PaintTree.cpp:16]: yeah thats  a good point. The last one is unreachable. As to what it represents, no idea will have to implement more to find out i think.

**Review by AaronVanGeffen — APPROVED** [review]
All appears to be working properly. Nice work.

**Pull request merged**